### ISSUE TITLE
Developer discussion

### ISSUE THREAD

**Issue by ScrappingFTW** [DoctorD1501/JAVMovieScraper#25]
Hi DoctorD,

I'm not really a java developer and trying to learn and need to clarify some concepts from the code. 

One thing I would like to understand is why you're using both Actions and ActionListeners for the controller layer instead of choosing one of them and stick with it.

I do see the convenience of having a single set of actions and share them between menuitems and buttons. So here's my proposal: moving all actionlisteners to actions, then initialize main menu and/or buttons with these actions.

Tell me what do you think about it and I'll submit a patch in a few days.

**Comment by geraldrotation** [comment]
Hi @DoctorD1501, I found this repo online while looking for a nice renamer for my huge jav library. And I must admit that this is some fine work you have done here. As a matter of fact, I've worked on something similar before but it could parse from dmm. So I'm wondering if there is anything that I could contribute to.

**Comment by DoctorD1501** [comment]
Hey Geraldrotation,

If you'd like to contribute and you don't have something of your own in mind, you can just take a look at the items on the issues page. Some of the tasks on there which might be a good thing for a beginner to this code base to do would be to create a new scraper. As long as you create the scraper in SiteParsingProfile.specific package and extend SiteParsingProfile and implement SpecificProfile (similar to the other scrapers in that package), you won't even have to worry about writing any GUI code or understanding how that part works too much.

If you know a lot about Java Swing or user interface design, you can also work on improving the interface or fixing various bugs in the program with the GUI.

**Comment by geraldrotation** [comment]
@DoctorD1501 
That sounds interesting. Another thing I was thinking of was to rewrite some of the core code to let them run concurrently. Right now, the scraping process can be slow when fetching from various websites. I will have a look at that.

**Comment by DoctorD1501** [comment]
That might work well if you can get it to work. The JAV movie amalgamation process actually runs 5 or 6 all at once, so there are some examples of getting that to work using different threads. The code to run that part of scraping is in a different method than the regular SiteSpecificParsing scraping, so there's definitely some work to be done there in bringing it all together in one unified process. One thing you may want to watch out for when testing is getting blocked by Google for scrapers that use that to make their search results. If you make too many requests too fast, you get banned for a long period of time and if you run 10-15 requests all at once, the ban could be easily triggered.

**Comment by DoctorD1501** [comment]
Closing this issue. If we want to have more discussions later, we can open a new issue.
